src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1985,7 +1985,7 @@ void MulticopterPositionControl::control_auto()
 			/* check if we just want to stay at current position */
 			matrix::Vector2f pos_sp_diff((_curr_pos_sp(0) - _pos_sp(0)), (_curr_pos_sp(1) - _pos_sp(1)));
 			bool stay_at_current_pos = (_pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LOITER
-						    || !next_setpoint_valid)
+						    || !next_setpoint_valid || _pos_sp_triplet.current.time_inside > 0.0f)
```

Comment:
Could you make one bool for all three cases?

Reply:
sure. I wanted to do the same since it is used more than once. On the other hand, I thought this way it is more clear what has changed and in another PR make a boolean out of it.